lib/api/efd.js

```diff
@@ -1,25 +1,13 @@
-import { InfluxDB } from "@influxdata/influxdb-client";
+import useSWR from "swr";
 
-export default function queryEfd(query) {
-  const url = process.env.NEXT_PUBLIC_EFD_URL;
-  const bucket = process.env.NEXT_PUBLIC_EFD_BUCKET;
-  const token = process.env.NEXT_PUBLIC_EFD_TOKEN;
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
-  const influxDB = new InfluxDB({ url, token });
-  const queryApi = influxDB.getQueryApi("");
+export default async function useEfd(type) {
+  const { data, error } = useSWR(`/api/efd/${type}`, fetcher);
 
-  return new Promise((resolve, reject) => {
-    const data = [];
-    queryApi.queryRows(query(bucket), {
-      next(row, tableMeta) {
-        data.push(tableMeta.toObject(row));
-      },
-      error(error) {
-        reject(error);
-      },
-      complete() {
-        resolve(data);
-      },
-    });
-  });
+  return {
```

Comment:
Becuase the env vars and the infuxing all lives in the `pages/api/efd` logic, this function is now only responsible for making the fetch to our internal API endpoint, and based on [some patterns I was reading about ](https://nextjs.org/docs/pages/building-your-application/data-fetching/client-side#client-side-data-fetching-with-swr), it seemed like I could use SWR here?